components/script/dom/cookiestore.rs

```diff
@@ -0,0 +1,568 @@
+
+/// <https://cookiestore.spec.whatwg.org/#create-a-cookielistitem>
+fn cookie_to_list_item(cookie: Cookie) -> CookieListItem {
+    CookieListItem {
+        // Let domain be the result of running UTF-8 decode without BOM on cookie’s domain.
+        domain: cookie
+            .domain()
+            .map(|domain| Some(domain.to_string().into())),
+
+        // Let expires be cookie’s expiry-time (as a timestamp).
+        expires: match cookie.expires() {
+            None | Some(cookie::Expiration::Session) => None,
+            Some(DateTime(time)) => Some(Some(Finite::wrap((time.unix_timestamp() * 1000) as f64))),
+        },
+
+        // Let name be the result of running UTF-8 decode without BOM on cookie’s name.
+        name: Some(cookie.name().to_string().into()),
+
+        // Let partitioned be a boolean indicating that the user agent supports cookie partitioning and that i
+        // that cookie has a partition key.
+        partitioned: Some(false), // Do we support partitioning? Spec says true only if UA supports it
+
+        // Let path be the result of running UTF-8 decode without BOM on cookie’s path.
+        path: cookie.path().map(|path| path.to_string().into()),
+
+        sameSite: match cookie.same_site() {
+            Some(SameSite::None) => Some(CookieSameSite::None),
+            Some(SameSite::Lax) => Some(CookieSameSite::Lax),
+            Some(SameSite::Strict) => Some(CookieSameSite::Strict),
+            None => None, // The spec doesnt handle this case, which implies the default of Lax?
+        },
+
+        // Let secure be cookie’s secure-only-flag.
+        secure: cookie.secure(),
+
+        // Let value be the result of running UTF-8 decode without BOM on cookie’s value.
+        value: Some(cookie.value().to_string().into()),
+    }
+}
+
+impl CookieStoreMethods<crate::DomTypeHolder> for CookieStore {
+    /// <https://cookiestore.spec.whatwg.org/#dom-cookiestore-get>
+    fn Get(&self, name: USVString, can_gc: CanGc) -> Rc<Promise> {
+        // 1. Let settings be this’s relevant settings object.
+        let global = self.global();
+
+        // 2. Let origin be settings’s origin.
+        let origin = global.origin();
+
+        // 5. Let p be a new promise.
+        let p = Promise::new(&global, can_gc);
+
+        // 3. If origin is an opaque origin, then return a promise rejected with a "SecurityError" DOMException.
+        if !origin.is_tuple() {
+            p.reject_error(Error::Security, can_gc);
+            return p;
+        }
+        // 4. Let url be settings’s creation URL.
+        let creation_url = global.creation_url();
+
+        self.in_flight.borrow_mut().push_back(p.clone());
+
+        // 6. Run the following steps in parallel:
+        let _ = self
+            .global()
+            .resource_threads()
+            .send(CoreResourceMsg::GetCookieDataForUrlAsync(
+                self.store_id,
+                creation_url.clone(),
+                Some(name.to_string()),
```

Comment:
Note: I'm not quite sure how to make this work with the decoding that happens later in parallel. Should we do the decoding before turning it into a string, which means decoding it here, or can we do it later on the string in parallel. But rust strings are already utf8 I believe. Needs some looking into. Any relevant tests failing at this point?

Reply:
I believe the only encoding/decoding tests that exist are checking if we strip the BOM from the name and we currently pass the test. (We don't strip it)

Reply:
Ok, I guess we can then leave it as it, and revisit when the others parts are implemented.